File: backend/src/main/java/com/staccato/visit/repository/VisitRepository.java
package com.staccato.visit.repository;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

import com.staccato.travel.domain.Travel;
import com.staccato.visit.domain.Visit;

public interface VisitRepository extends JpaRepository<Visit, Long> {
    List<Visit> findAllByTravelIdOrderByVisitedAt(long travelId);

    boolean existsByTravel(Travel travel);

    void deleteAllByTravelId(long travelId);
}

File: android/Staccato_AN/app/src/main/java/com/woowacourse/staccato/domain/repository/VisitRepository.kt
package com.woowacourse.staccato.domain.repository

import com.woowacourse.staccato.data.dto.moment.MomentCreationResponse
import com.woowacourse.staccato.domain.model.Visit
import okhttp3.MultipartBody
import java.time.LocalDateTime

interface VisitRepository {
    suspend fun getVisit(visitId: Long): Result<Visit>

    suspend fun createVisit(
        memoryId: Long,
        placeName: String,
        latitude: String,
        longitude: String,
        address: String,
        visitedAt: LocalDateTime,
        visitImageMultiParts: List<MultipartBody.Part>,
    ): Result<MomentCreationResponse>

    suspend fun updateVisit(
        visitId: Long,
        placeName: String,
        visitImageUrls: List<String>,
        visitImageMultiParts: List<MultipartBody.Part>,
    ): Result<Unit>

    suspend fun deleteVisit(visitId: Long): Result<Unit>
}
